src/Request.php

```diff
@@ -156,7 +156,7 @@ public function handleData($data)
 
             $this->emit('response', array($response, $this));
 
-            $response->emit('data', array($bodyChunk, $response));
+            $this->stream->emit('data', array($bodyChunk, $response));
```

Comment:
Check you're right it should be omitted there, the data will flow through the stream to the response and the response will emit it again when data comes from the stream :+1: